github-actions/post-approval-changes/lib/main.ts

```diff
@@ -131,6 +110,25 @@ async function runPostApprovalChangesAction(client: Octokit): Promise<void> {
   });
 }
 
+/** Set of membership lookup results, used as cache for lookups. */
+const isGooglerOrgMemberCache = new Map<string, boolean>([]);
+
+async function isGooglerOrgMember(client: Octokit, username: string): Promise<boolean> {
+  if (isGooglerOrgMemberCache.has(username)) {
+    return isGooglerOrgMemberCache.get(username)!;
+  }
+  return await client.orgs
+    .checkMembershipForUser({org: 'googler', username})
+    .then(
+      ({status}) => (status as number) === 204,
+      () => false,
+    )
+    .then((result) => {
+      isGooglerOrgMemberCache.set(username, result);
+      return result;
```

Comment:
Nit: Why not use await/try/finally like we do in most places in the repo?

Reply:
I think it ends up less readable in this case:

```ts
async function isGooglerOrgMember(client: Octokit, username: string): Promise<boolean> {
  if (isGooglerOrgMemberCache.has(username)) {
    return isGooglerOrgMemberCache.get(username)!;
  }
  let result: boolean;
  try {
    result = await client.orgs.checkMembershipForUser({org: 'googler', username}).then(
      ({status}) => (status as number) === 204,
      () => false,
    );
  } catch {
    result = false;
  } finally {
    isGooglerOrgMemberCache.set(username, result!);
    return result!;
  }
}
```

Reply:
I think it could be simplified further, but that's not a big deal either.